R/robustbayesianmetaanalysis.R

```diff
@@ -729,571 +577,456 @@ RobustBayesianMetaAnalysis <- function(jaspResults, dataset, options, state = NU
-      conditionalStudiesSummary$addFootnote(gettextf("Estimated studies effects (%s) are model averaged over models assuming existence of effect.", "\u03B8"))
-      mainSummary[["conditionalStudiesSummary"]] <- conditionalStudiesSummary
+
+    # PET-PEESE table
+    if (any(grepl("PET", rownames(fitSummary[["estimates_conditional"]])) | grepl("PEESE", rownames(fitSummary[["estimates_conditional"]])))) {
+      conditionalPetPeese <- createJaspTable(title = gettextf("Conditional PET-PEESE Estimates"))
+      conditionalPetPeese$position <- 7
+      conditionalPetPeese <- .robmaTableFillPetPeese(conditionalPetPeese, fitSummary[["estimates_conditional"]], options)
+      mainSummary[["conditionalPetPeese"]] <- conditionalPetPeese
     }
-    
+
   }
-  
+
   return()
 }
 .robmaModelsOvervievTable      <- function(jaspResults, options) {
+
   # extract the model
   fit   <- jaspResults[["model"]][["object"]]
-  
+
   # some shared info
-  fitSummary <- RoBMA::summary.RoBMA(
+  fitSummary <- summary(
     fit,
-    type     = "models",
-    logBF    = options[["bayesFactorType"]] == "LogBF10",
-    BF01     = options[["bayesFactorType"]] == "BF01",
-    probs    = c(.5 + c(-1, 1) * options[["resultsCI"]] / 2)
+    type       = "models",
+    short_name = options[["shortNames"]]
   )
-  
+
   # do ordering
-  if (options[["resultsModelsOrder"]] == "marglik") {
-    fitSummary[["overview"]] <- fitSummary[["overview"]][order(fitSummary[["overview"]][["log(MargLik)"]], decreasing = TRUE),]
-  } else if (options[["resultsModelsOrder"]] == "posterior") {
-    fitSummary[["overview"]] <- fitSummary[["overview"]][order(fitSummary[["overview"]][["Post. prob."]], decreasing = TRUE),]
-  }
-  
+  if (options[["resultsModelsOrder"]] == "marglik")
+    fitSummary[["summary"]] <- fitSummary[["summary"]][order(fitSummary[["summary"]][["marglik"]], decreasing = TRUE),]
+  else if (options[["resultsModelsOrder"]] == "posterior")
+    fitSummary[["summary"]] <- fitSummary[["summary"]][order(fitSummary[["summary"]][["post_prob"]], decreasing = TRUE),]
+
   # compute the BF requested
-  if (options[["resultsModelsBF"]] == "inclusion") {
-    bf <- fitSummary[["overview"]][, 7]
-  } else if (options[["resultsModelsBF"]] == "best") {
-    bf <- exp(fitSummary[["overview"]][["log(MargLik)"]]) / exp(max(fitSummary[["overview"]][["log(MargLik)"]]))
-  } else if (options[["resultsModelsBF"]] == "previous") {
-    tempThisMargLik <- exp(fitSummary[["overview"]][["log(MargLik)"]])[-length(fitSummary[["overview"]][["log(MargLik)"]])]
-    tempPrevMargLik <- exp(fitSummary[["overview"]][["log(MargLik)"]])[-1]
+  if (options[["resultsModelsBf"]] == "inclusion") {
+    bf <- fitSummary[["summary"]][, 7]
+  } else if (options[["resultsModelsBf"]] == "best") {
+    bf <- exp(fitSummary[["summary"]][["marglik"]]) / exp(max(fitSummary[["summary"]][["marglik"]]))
```

Comment:
Instead of `exp(a) / exp(b)` do `exp(a-b)` to avoid over and underflow. Just compare `exp(710) / exp(709)` to `exp(710 - 709)`.

Reply:
thanks, a good point!